pkg/model/awsmodel/autoscalinggroup.go

```diff
@@ -74,19 +74,23 @@ func (b *AutoscalingGroupModelBuilder) Build(c *fi.CloudupModelBuilderContext) e
 			}
 		}
 
-		task, err := b.buildLaunchTemplateTask(c, name, ig)
+		userData, err := b.BootstrapScriptBuilder.ResourceNodeUp(c, ig)
 		if err != nil {
 			return err
 		}
-		c.AddTask(task)
```

Comment:
Looks like a good fix.  But why are we still building userData outside the if statement - it looks like we don't need it if `ig.Spec.Manager == "Karpenter" `?  Is this for error checking?  (If so, that makes sense, but let's add a comment to protect it from linting)